### PR TITLE
add preprocess function in commands

### PR DESCRIPTION
**Pull request by Liujingfang1**
This change adds the following
- add a preprocess function so that the inventoryPolicy can be adjusted
- add a function in InventoryClient to update the inventory object labels if the cluster inventory object exists

This change should be merged after #313

### CODE REVIEW

File: pkg/inventory/inventory-client.go [hunk 1]
```diff
@@ -268,6 +272,18 @@ func (cic *ClusterInventoryClient) getClusterInventoryInfo(localInv *unstructure
 	return clusterInv, nil
 }
 
+func (cic *ClusterInventoryClient) UpdateLabels(inv InventoryInfo, labels map[string]string) error {
```
Comment (seans3) [inline pkg/inventory/inventory-client.go:275]: Should we add a unit test for this new function?

File: pkg/apply/destroyer.go [hunk 2]
```diff
@@ -75,6 +79,7 @@ func (d *Destroyer) Run(inv inventory.InventoryInfo) <-chan event.Event {
 		err := d.PruneOptions.Prune(inv, nil, sets.NewString(), taskContext, prune.Options{
 			DryRunStrategy:    d.DryRunStrategy,
 			PropagationPolicy: metav1.DeletePropagationBackground,
+			InventoryPolicy:   option.InventoryPolicy,
```
Comment (seans3) [inline pkg/apply/destroyer.go:82]: Should there be unit tests with different inventory policies for `Destroyer` now?

File: cmd/preview/cmdpreview.go [hunk 3]
```diff
@@ -91,6 +98,11 @@ func (r *PreviewRunner) RunE(cmd *cobra.Command, args []string) error {
 		r.Destroyer.DryRunStrategy = drs
 	}
 
+	inventoryPolicy, err := flagutils.ConvertInventoryPolicy(r.inventoryPolicy)
```
Comment (seans3) [inline cmd/preview/cmdpreview.go:101]: Should there be new unit tests (or other tests) for the new `InventoryPolicy`?

File: cmd/flagutils/utils.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+package flagutils
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/cli-utils/pkg/inventory"
+)
+
+func ConvertInventoryPolicy(policy string) (inventory.InventoryPolicy, error) {
```
Comment (seans3) [inline cmd/flagutils/utils.go:12]: Should there be a simple test for this new utility function?

**Review by seans3 — CHANGES_REQUESTED** [review]
Maybe we should add a few new tests to ensure this new `InventoryPolicy` is working correctly.

File: pkg/inventory/inventory-client.go (hunk 1 above)
Comment (Liujingfang1) [reply pkg/inventory/inventory-client.go:275]: Filed #317 for this. We'll address it in a future PR.

File: cmd/flagutils/utils.go (hunk 4 above)
Comment (Liujingfang1) [reply cmd/flagutils/utils.go:12]: Done

File: cmd/preview/cmdpreview.go (hunk 3 above)
Comment (Liujingfang1) [reply cmd/preview/cmdpreview.go:101]: We'll address this in future PRs.

File: pkg/apply/destroyer.go (hunk 2 above)
Comment (Liujingfang1) [reply pkg/apply/destroyer.go:82]: We'll address this in future PRs.

**Review by seans3 — APPROVED** [review]
/lgtm

**Pull request merged**